teps/0121-refine-retries-for-taskruns-and-customruns.md

```diff
@@ -78,6 +79,201 @@ They set both `retries` and `timeout` for the two resources, under the current i
 - `task-run-example` will be retried once after 10s.
 - `custom-run-example` will be timed out after 10s. But if the corresponding CustomRun controller implements retries **for *each* attempt**, like in TaskRuns, instead of **for all attempts** per the documented guidance, then the `custom-run-example` would be retried once after 10s, working similarly to the `task-run-example`.
 
+#### Retries in Custom Runs
+
+Some Custom Tasks need retry logic that is different from just creating a new instance of a Run when a previous attempt has failed, as described in [TEP-0069](./0069-support-retries-for-custom-task-in-a-pipeline.md).
+One example is the [TaskLoop custom task](https://github.com/tektoncd/experimental/tree/main/task-loops), which creates one TaskRun for each value in an array parameter for a single Task.
+If retries are specified, only failed TaskRuns are retried. Creating a new Run would retry every TaskRun, even the successful ones.
+
+```yaml
+apiVersion: custom.tekton.dev/v1alpha1
+kind: TaskLoop
+metadata:
+  name: testloop
+spec:
+  taskRef:
+    name: my-task
+  iterateParam: test-type
+  retries: 2
+```
+
+```yaml
+apiVersion: tekton.dev/v1alpha1
+kind: Run
+metadata:
+  generateName: testloop-run-
+spec:
+  params:
+    - name: test-type
+      value:
+        - codeanalysis
+        - unittests
+        - e2etests
+  ref:
+    apiVersion: custom.tekton.dev/v1alpha1
+    kind: TaskLoop
+    name: testloop
+```
+
+When this Run is created by itself, it will create three TaskRuns, and retry any that fail twice.
+
+It's also possible to specify retries on the Run, for example:
+
+By reference:
+```yaml
+apiVersion: tekton.dev/v1alpha1
+kind: Run
+metadata:
+  generateName: testloop-run-
+spec:
+  params:
+    - name: test-type
+      value:
+        - codeanalysis
+        - unittests
+        - e2etests
+  retries: 2
+  ref:
+    apiVersion: custom.tekton.dev/v1alpha1
+    kind: TaskLoop
+    name: testloop
+```
+
+Inline:
+```yaml
+apiVersion: tekton.dev/v1alpha1
+kind: Run
+metadata:
+  generateName: testloop-run-
+spec:
+  params:
+    - name: test-type
+      value:
+        - codeanalysis
+        - unittests
+        - e2etests
+  retries: 5
+  spec:
+    apiVersion: custom.tekton.dev/v1alpha1
+    kind: TaskLoop
+    spec:
+      taskRef:
+        name: my-task
+      iterateParam: test-type
+      retries: 2
+```
+In both cases, retries are defined on both the Run and in the custom task spec, resulting in undefined behavior.
+It's up to the custom task controller to decide what to do in this case.
```

Comment:
If a custom run implementor decided to give an authoring time setting for retries, i.e. in your example a `retries` field in the `TaskLoop`, I think it should follow the same approach that we have for other fields that can be defined at both author and runtime, i.e. runtime takes precedence. 

We could add the following in the custom task docs: "If a custom task controller implements retries capabilities, it **MUST**  honour the `retries` field in the `CustomRun`. Adding a `retries` field in the custom resources is discouraged and when provided, the `retries` field in the `CustomRun` **MUST** take precedence."